newrelic/core/super_agent_health.py

```diff
@@ -115,88 +132,81 @@ def health_check_enabled(self):
             return False
 
         health_file_location = os.environ.get("NEW_RELIC_SUPERAGENT_HEALTH_DELIVERY_LOCATION", None)
-        valid_file_location = is_valid_file_delivery_location(health_file_location)
-        if not valid_file_location:
-            return False
 
-        return True
+        return is_valid_file_delivery_location(health_file_location)
+
+    @property
+    def is_healthy(self):
+        return True if self.status == "Healthy" else False
 
     def set_health_status(self, health_status, response_code=None, info=None):
+        license_key_error = True if self.status == "License key missing in configuration" or self.status == "Invalid license key (HTTP status code 401)" else False
```

Comment:
I don't think we can be comparing the previous status to the current one based on the text of the error message, that's both a slow operation and also incredibly fragile. If anyone updates the messages up above it'll introduce bugs unless they happen to spot that this message is duplicated and used in logic.

Can we store either the `license_key_error = True/False` on `self`, or store the `health_status` enum value on the `self`? I would think the latter would be more useful and general.